script.js

```diff
@@ -1,4 +1,149 @@
+    drawCount += 1;
+  } else if (
+    (compChoice == "scissors" && input == "paper") ||
+    (compChoice == "paper" && input == `stone`) ||
+    (compChoice == `stone` && input == `scissors`)
+  ) {
+    myOutputValue = "The computer wins :(";
+    compWins += 1;
+  } else if (
+    (compChoice == "scissors" && input == `stone`) ||
+    (compChoice == "paper" && input == `scissors`) ||
+    (compChoice == `stone` && input == "paper")
+  ) {
+    myOutputValue = "You win!";
+    userWins += 1;
+    // the below determines who wins if the computer rolls a 2, i.e. paper
+  } else {
+    myOutputValue = `${userName}, you have entered something that isn't recognized! Please enter either 'scissors', 'paper', or 'stone'.`;
+  }
+};
+
+var reverseGame = function (compChoice, input) {
+  // if reverse game is true
+  if (
+    (compChoice == "scissors" && input == `reversed scissors`) ||
+    (compChoice == "paper" && input == "reversed paper") ||
+    (compChoice == `stone` && input == `reversed stone`)
+  ) {
+    //below is for the reverse game
+    // the below determines who wins if the computer rolls a 1, i.e. scissors
+    myOutputValue = "It's a draw ...";
+    drawCount += 1;
+  } else if (
+    (compChoice == "scissors" && input == "reversed paper") ||
+    (compChoice == "paper" && input == `reversed stone`) ||
+    (compChoice == `stone` && input == `reversed scissors`)
+  ) {
+    myOutputValue = "You win!";
+    userWins += 1;
+  } else if (
+    (compChoice == "scissors" && input == `reversed stone`) ||
+    (compChoice == "paper" && input == `reversed scissors`) ||
+    (compChoice == `stone` && input == "reversed paper")
+  ) {
+    myOutputValue = "The computer wins :(";
+    compWins += 1;
+    // the below determines who wins if the computer rolls a 2, i.e. paper
+  } else {
+    myOutputValue = `${userName}, you have entered something that isn't recognized! Please enter either 'scissors', 'paper', or 'stone'.`;
+  }
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (userName == "") {
+    userName = input;
+    return `Hello ${userName}. Please enter either 'scissors', 'paper' or 'stone' to start the game!`;
+  }
+  var reverseGameMode;
+  //below is to divert the fork for the reverse game or the normal game
+  if (
+    input == "reversed scissors" ||
+    input == "reversed paper" ||
+    input == "reversed stone"
+  ) {
+    reverseGameMode = true;
+  } else if (input == "scissors" || input == "paper" || input == "stone") {
+    reverseGameMode = false;
+  } else {
+    myOutputValue = "please type in either 'scissors', 'paper', or 'stone'";
+  }
+  if (reverseGameMode == false) {
+    normalGame(compChoice(rollDice()), input);
+  } else if (reverseGameMode == true) {
+    reverseGame(compChoice(rollDice()), input);
+    myOutputValue =
+      myOutputValue +
+      " Sneaky! You have played reverse scissors, paper, stone!";
+    //return finalString;
```

Comment:
I see that you are using the helper functions to change the global variables directly. This might be sneaky as it may be hard to check which functions are working and what they are doing etc. 
It would be easier to track if: 
input>function>output> output variable that does things